pkgs/applications/virtualization/qemu/default.nix

```diff
@@ -143,7 +143,12 @@ stdenv.mkDerivation rec {
     # will still be needed in nix until the macOS SDK reaches 10.13+.
     ./provide-fallback-for-utimensat.patch
   ]
-    ++ lib.optional nixosTestRunner ./force-uid0-on-9p.patch;
+    ++ lib.optional nixosTestRunner ./force-uid0-on-9p.patch
+    # Support Linux 5.17 guests on aarch64-darwin. Remove for v7.0.
+    ++ lib.optional (stdenv.isDarwin && stdenv.isAarch64) (fetchpatch {
```

Comment:
When applying patches, please do so unconditionally, unless that would cause problems for other platforms.  That way, everybody working on the package will be working with the same set of patches, so won't e.g. fail to notice if a patch for another platform no longer applies.

It's still okay to target master, even if the patch is applied for all platforms, because QEMU has very few dependents.